core/src/banking_stage/forwarder.rs

```diff
@@ -65,15 +102,14 @@ impl Forwarder {
         // load all accounts from address loader;
         let current_bank = self.bank_forks.read().unwrap().working_bank();
 
-        let mut forward_packet_batches_by_accounts =
-            ForwardPacketBatchesByAccounts::new_with_default_batch_limits();
+        self.clear_batches();
```

Comment:
let's move this to the end, so we clear the batches after each forwarding.
otherwise the packets are held for longer than necessary between forward calls

Reply:
moved for both scheduler, eye balled there aren't early exits before `clear_batches()`.